### PR TITLE
complete task 4

### PR DESCRIPTION
**Pull request by jesjin**
Complete task 4

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -42,7 +42,47 @@ describe('GET /api', () => {
...
+        .then(({ body }) => {
+          const expectedShape = {
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: 1,
+            body: expect.any(String),
+            topic: expect.any(String),     
+            created_at: expect.any(String),          
+            votes: expect.any(Number),
+            article_img_url: expect.any(String)
```
Comment (nc-linda-guester) [inline __tests__/app.test.js:65]: A bit pedantic, but if we are getting one specific article, we know what the title, topic, author etc. are looking like and can be more specific about what we want to get back

File: __tests__/app.test.js [hunk 2]
```diff
@@ -42,7 +42,47 @@ describe('GET /api', () => {
...
+            expect(body.msg).toBe('Bad Request - Article ID was invalid.');
+        });
+    });
+    test('404: responds with "Article not found" for non-existent article ID', () => {
+        return request(app)
+        .get('/api/articles/999')
+        .expect(404)
+        .expect('Content-Type', /json/)
+        .then(({ body }) => {
+            expect(body.msg).toBe('Article not found');
```
Comment (nc-linda-guester) [inline __tests__/app.test.js:84]: Well done thinking of the tests needed!

File: app.js [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
 const app = express();
 
+app.get('/api/articles/:article_id', getArticleById);
 app.get('/api/topics', getTopics);
 app.get('/api', getApiEndPoints);
 
 app.use((err, req, res, next) => {
   if (err.code === '22P02') {
     res.status(400).send({ msg: 'Bad Request' });
+  } else if (err.status) {
```
Comment (nc-linda-guester) [inline app.js:15]: If you specify on line 16 you want to send the err.msg, it's generally considered good form to include this in the else if - e.g. `(err.status && err.msg)`

File: controllers/articleController.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const { getArticleById } = require('../models/articleModel');
+
+exports.getArticleById = (req, res, next) => {
+  const { article_id } = req.params;
+  // check that article id is an integer, if it is not, return an error
+  if (!Number.isInteger(Number(article_id))) {
+    return res.status(400).send({ msg: 'Bad Request - Article ID was invalid.' });
+  }
+
```
Comment (nc-linda-guester) [inline controllers/articleController.js:9]: You already added an error handler for PSQL errors that you are not using - personally, I would say change your test to the error message used in your PSQL error handling & delete this, makes for cleaner code :)

**Review by nc-linda-guester — COMMENTED** [review]
Seems like there is a bit of confusion going on with error handling - have a look at the notes: https://notes.northcoders.com/courses/js-back-end/error-handling

Apart from that, well done! 😄

File: controllers/articleController.js (hunk 4 above)
Comment (jesjin) [reply controllers/articleController.js:9]: I took this approach to protect against SQL injection

**Pull request merged**